mumax3c/scripts/util.py

```diff
@@ -81,22 +81,49 @@ def set_parameter(parameter, name, system):
 
     # Spatially varying parameter defined using subregions.
     elif isinstance(parameter, dict):
-        names = system.m.mesh.subregions.keys()
-        subregions_dict = dict(zip(names, range(len(names))))
-
         for key, value in parameter.items():
-            if isinstance(value, numbers.Real):
-                mx3 += f"{name}.setregion({subregions_dict[key]}, {value})\n"
-
-            elif isinstance(value, (list, tuple, np.ndarray)):
-                mx3 += (
-                    f"{name}.setregion({subregions_dict[key]}, "
-                    "vector({}, {}, {}))\n".format(*value)
-                )
+            if ":" in key:
+                mx3 += _set_inter_reg_params(key, value, name, system)
+            else:
+                for region in system.region_relator[key]:
```

Comment:
Do we need to set interregion coupling for the different mumax regions that share the same material parameter (e.g. A)?

Reply:
I think mumax takes an average of `Aex` when they are adjacent (correct me if I am wrond @samjrholt). So technically we do not need to set it. But chances are low that a user will define `A_dict = {"r1": A, "r2": A, "r1:r2": A}`.

Reply:
What about two mumax regions (different Ms) in `"r1"` (or just a constant value for the whole sample) with value `A1`(and mabye `"r2": A2`)?

Reply:
Good point, yes I think we should probably be explicit about interactions within `subregions` but between different `mumax3` regions

Reply:
> What about two mumax regions (different Ms) in `"r1"` (or just a constant value for the whole sample) with value `A1`(and mabye `"r2": A2`)?

In this case, the different "mumax regions" will be coupled with an average of exchange which is same and `"r2": A2` should work fine similarly. However, due to the current state, any `"r1:r2": A3` should fail since `r1` hase more than one "mumax regions".

Reply:
Also, the average of exchange between two adjacent regions is [here](https://groups.google.com/g/mumax2/c/284kRz2QdPA/m/4laXosuiAQAJ)